File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int SumNumbers(TreeNode root) {
        return SumNumbersHelper(root, 0);
    }

    private int SumNumbersHelper(TreeNode node, int currentSum) {
        if (node == null) {
            return 0;
        }

        currentSum = currentSum * 10 + node.val;

        // If it's a leaf node, return the current sum
        if (node.left == null && node.right == null) {
            return currentSum;
        }

        // Recursively calculate the sum for left and right subtrees
        int leftSum = SumNumbersHelper(node.left, currentSum);
        int rightSum = SumNumbersHelper(node.right, currentSum);

        return leftSum + rightSum;
    }
}

File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    const helper =(node, currentSum)=>{
        if (!node)return 0;
        currentSum=currentSum*10+node.val;
        if(!node.left && !node.right){return currentSum;}
        return helper(node.left,currentSum)+helper(node.right,currentSum);
    };
    return helper(root, 0);
};